lib/ansible/modules/messaging/rabbitmq/rabbitmq_plugin.py

```diff
@@ -147,11 +154,15 @@ def main():
     state = module.params['state']
 
     rabbitmq_plugins = RabbitMqPlugins(module)
-    enabled_plugins = rabbitmq_plugins.get_all()
+    enabled_plugins, available_plugins = rabbitmq_plugins.get_all()
 
     enabled = []
     disabled = []
     if state == 'enabled':
+        for name in names:
+            if name not in available_plugins:
+                module.fail_json(msg='{name} plugin is not available'.format(name=name))
+
         if not new_only:
```

Comment:
Should this only apply when enabling plugins or shoud this check be applied to disabling as well?

Might be worth moving this check to line 158 so it's outside the `'if state =='enabled'` block so that we check the plugins actually exist regardless of weather we're enabling or disabling.

I'm not a rabitMQ user so there may be a good reason I'm unaware of to apply only to enabling in which case I'd suggest a comment is worthwhile.

Reply:
Well, my thought on this is: if you want the plugin to be disabled, and it's missing - then you get your target state: plugin is not enabled.